src/components/modules/blockManager.ts

```diff
@@ -667,6 +667,18 @@ export default class BlockManager extends Module {
     this.Editor.UI.checkEmptiness();
   }
 
+  /**
+   * Cleans up all the block tools' resources
+   * This is called when editor is destroyed
+   */
+  public destroy(): void {
+    this.blocks.forEach((block) => {
```

Comment:
Using `await Promise.all(this.blocks.map(block) => { /*further code*/ })` would be more accurate as it will cover async tasks if present in any of the `destroy` functions.

Reply:
Sure I'll do it tonight (Saturday night IST). That'll require a chain of async functions I rekon. I'll have a look at it and get back to you.

Reply:
Please also add your changes to the CHANGELOG.md file with the link to the issue.